File: leden/sawny.py
from sys import stdin

# -20 meter = 90s     +30s
# -5  meter = 60s     +0s
# 0   meter = 70s     +10s
# 10  meter = 90s     +20s

data      = stdin.read().split()
length    = data.pop(0);
wayPoints = map(int, data);

time      = 0;
stdSlope  = -5;
baseTime  = 60;

for i, wayPoint in enumerate(wayPoints[0:-1]):
    slope = -1*(wayPoint-wayPoints[i+1]);
    time  = time + abs(stdSlope - slope)*2 + baseTime;    

print time;

File: leden/sawny.hs
import System.IO (getLine)

toInt :: [String] -> [Int]
toInt = map read

time :: (Num a) => [a] -> a
time (x:y:xs)  = abs(5 - (x-y))*2+60 + time (y:xs)
time otherwise = 0

main = do
    -- Get the data from stdin and process it to a [Int]
    _     <- getLine
    stdin <- getLine
    let wayPoints = toInt $ words stdin

    -- Solve the problem
    print $ time wayPoints